src/Console/Command/SelfUpdateCommand.php

```diff
@@ -0,0 +1,205 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the humbug/php-scoper package.
+ *
+ * Copyright (c) 2017 Théo FIDRY <theo.fidry@gmail.com>,
+ *                    Pádraic Brady <padraic.brady@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Humbug\PhpScoper\Console\Command;
+
+use Humbug\SelfUpdate\Updater;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+
+final class SelfUpdateCommand extends Command
+{
+    /** @internal */
+    const REMOTE_FILENAME = 'php-scoper.phar';
+    /** @internal */
+    const STABILITY_STABLE = 'stable';
+    /** @internal */
+    const PACKAGIST_PACKAGE_NAME = 'humbug/php-scoper';
+    /** @internal */
+    const ROLLBACK_OPT = 'rollback';
+    /** @internal */
+    const CHECK_OPT = 'check';
+
+    /**
+     * @var OutputInterface
+     */
+    private $output;
+
+    /**
+     * @var string
+     */
+    private $version;
+
+    protected function configure()
+    {
+        $this
+            ->setName('self-update')
+            ->setDescription('Update php-scoper.phar to most recent stable build.')
```

Comment:
can the name be changed? I mean the user may have renamed `php-scoper.phar` to something else e.g. `php-scoper` or `scoper`